api/envoy/extensions/filters/http/router/v3/router.proto

```diff
@@ -21,11 +21,20 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // Router :ref:`configuration overview <config_http_filters_router>`.
 // [#extension: envoy.filters.http.router]
 
-// [#next-free-field: 9]
+// [#next-free-field: 10]
 message Router {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.http.router.v2.Router";
 
+  message UpstreamAccessLogOptions {
+    // If set to true, an upstream access log will be recorded when an upstream stream is
+    // associated to an http request. Note: Each HTTP request received for an already established
+    // connection will result in an upstream access log record. This includes, for example,
+    // consecutive HTTP requests over the same connection or a request that is retried.
+    // In case a retry is applied, an upstream access log will be recorded for each retry.
+    bool flush_upstream_log_on_new_request = 1;
```

Comment:
Should this be a boolean or an optional AccessLog message?  I could imagine folks wanting to send their connection established logs to a different endpoint than standard upstream access logs though I'll defer to @lizan as API reviewer for that.

mild preference for on_upstream_stream rather than new_request, as request/response is minimally used in Envoy

Reply:
> Should this be a boolean or an optional AccessLog message? I could imagine folks wanting to send their connection established logs to a different endpoint than standard upstream access logs though I'll defer to @lizan as API reviewer for that.

Is it not possible for the user to use the existing AccessLog field with another sink?

> mild preference for on_upstream_stream rather than new_request, as request/response is minimally used in Envoy

Makes sense, I'll update it